src/slice.cr

```diff
@@ -503,6 +503,21 @@ struct Slice(T)
     end
   end
 
+  # Returns an `Array` with the results of running the block against each element of the slice.
+  #
+  # ```
+  # slice = Slice.new(3, 10)
+  # slice.to_a { |elm| elm.to_s } # => ["10", "10", "30"]
+  # ```
+  def to_a(&block : T -> Q) forall Q
```

Comment:
I'm not sure if this method will get accepted unfortunately. Seems to be conflating two things. Not sure if there's an efficient way to do this already.

Leave it in for now until people discuss this.

Reply:
Just realized it can be done like so:

```
an_array = a_slice.to_a()
an_array.map!() { |e| e.to_s }
```

Reply:
Fixed

Reply:
This actually wont work because the type changes... Reverting.

Reply:
I dont know how else to achieve this without making a temporary array. Which is wasteful.

Reply:
You don't need `Slice#to_a`. Just create an `[] of String`, iterate the slice and populate the array.

Reply:
You can also do this easily with the Array.new(size, &block) method.

Reply:
Fixed